src/allotropy/parsers/msd_workbench/msd_workbench_structure.py

```diff
@@ -0,0 +1,168 @@
+                WellData.create(
+                    luminescence=row_series[int, "Signal"],
+                    location_id=row_series[str, "Well"] + "_" + row_series[str, "Spot"],
+                    sample_id=row_series[str, "Sample"] + "_" + row_series[str, "Well"],
+                    concentration=row_series[float, "Concentration"],
+                    measurement_custom_info={
+                        "detection range": row_series.get(str, "Detection Range"),
+                        "assay identifier": row_series.get(str, "Assay"),
+                    },
+                    dilution_factor=row_series.get(int, "Dilution Factor"),
+                )
+            )
+
+        well_column = data["Well"]
+        plate_well_count = len(well_column.unique())
+
+        return PlateData(
+            measurement_time=DEFAULT_EPOCH_TIMESTAMP,
+            well_plate_id=well_plate_id,
+            plate_well_count=plate_well_count,
+            well_data=well_data,
+        )
+
+
+@dataclass(frozen=True)
+class WellData:
+    luminescence: int
+    location_identifier: str
+    sample_identifier: str
+    sample_role_type: SampleRoleType | None
+    mass_concentration: float | None
+    dilution_factor: int | None = None
+    measurement_custom_info: dict[str, Any] | None = None
+
+    @staticmethod
+    def create(
+        luminescence: int,
+        location_id: str,
+        sample_id: str,
+        concentration: float | None = None,
+        measurement_custom_info: dict[str, Any] | None = None,
+        dilution_factor: int | None = None,
+    ) -> WellData:
+        return WellData(
+            luminescence=luminescence,
+            location_identifier=location_id,
+            sample_identifier=sample_id,
+            sample_role_type=SAMPLE_ROLE_TYPE_MAPPING.get(sample_id[0].lower()),
+            mass_concentration=concentration,
+            measurement_custom_info=measurement_custom_info,
+            dilution_factor=dilution_factor,
+        )
+
+
+def create_metadata(header: Header) -> Metadata:
+    asm_file_identifier = Path(header.file_name).with_suffix(".json")
+    return Metadata(
+        file_name=header.file_name.rsplit("\\", 1)[-1],
+        unc_path=header.file_name,
+        software_name=header.name,
+        device_identifier=NOT_APPLICABLE,
+        model_number=header.model,
+        asm_file_identifier=asm_file_identifier.name,
+        data_system_instance_id=NOT_APPLICABLE,
+    )
+
+
+def create_measurement_groups(plate_data: PlateData) -> list[MeasurementGroup]:
+    grouped_wells = defaultdict(list)
+
+    for well in plate_data.well_data:
+        grouped_wells[well.sample_identifier].append(well)
+
+    well_data: list[list[WellData]] = list(grouped_wells.values())
+
+    return [
+        MeasurementGroup(
+            measurement_time=plate_data.measurement_time,
+            plate_well_count=plate_data.plate_well_count,
+            measurements=[
+                Measurement(
```

Comment:
This seems like a pretty 1:1 mapping from fields in "Well" to the measurement fields, I think should should probably just create the measurements directly from rows (no need for intermediate classes - these are really only needed when there are multiple different sections of data that needs to be used to create a measurement)